src/ast/ast_validations.c

```diff
@@ -366,19 +512,40 @@ static AST_Validation _Validate_CREATE_Clause_TypedRelations(const cypher_astnod
     return AST_VALID;
 }
 
-static AST_Validation _Validate_CREATE_Clause(const AST *ast, char **reason) {
-    const cypher_astnode_t *create_clause = AST_GetClause(ast, CYPHER_AST_CREATE);
-    if (!create_clause) return AST_VALID;
+static AST_Validation _Validate_CREATE_Clause_Properties(const cypher_astnode_t *clause, char **reason) {
+	const cypher_astnode_t *pattern = cypher_ast_create_get_pattern(clause);
+	uint path_count = cypher_ast_pattern_npaths(pattern);
+	for (uint i = 0; i < path_count; i ++) {
+		const cypher_astnode_t *path = cypher_ast_pattern_get_path(pattern, i);
+		if (_ValidateInlinedPropertiesOnPath(path, reason) != AST_VALID) return AST_INVALID;
+	}
 
-    if (_Validate_CREATE_Clause_TypedRelations(create_clause) == AST_INVALID) {
-        asprintf(reason, "Exactly one relationship type must be specified for CREATE");
-        return AST_INVALID;
-    }
+	return AST_VALID;
+}
 
-    return AST_VALID;
+static AST_Validation _Validate_CREATE_Clauses(const AST *ast, char **reason) {
+    const cypher_astnode_t **create_clauses = AST_GetClauses(ast, CYPHER_AST_CREATE);
+    if (!create_clauses) return AST_VALID;
+
+	AST_Validation res = AST_VALID;
+	uint clause_count = array_len(create_clauses);
+	for(uint i = 0; i < clause_count; i++) {
+		if(_Validate_CREATE_Clause_TypedRelations(create_clauses[i]) == AST_INVALID) {
+			asprintf(reason, "Exactly one relationship type must be specified for CREATE");
+			res = AST_INVALID;
+			goto cleanup;
+		}
+		// Validate that inlined properties do not use parameters
+        res = _Validate_CREATE_Clause_Properties(create_clauses[i], reason);
```

Comment:
Why do we pass `reason` here but not for `_Validate_CREATE_Clause_TypedRelations` ?

Reply:
When `_Validate_CREATE_Clause_TypedRelations` fails, we always emit the error:
"Exactly one relationship type must be specified for CREATE"
Which is similar to Neo's response, as there's not much to provide the user with in the case of a query like `CREATE (:a)-[]->(:b)`.

For property validation, we can at least differentiate between whether the user is trying to use a parameter or a non-constant value, and as such can give a slightly more specific erro.